### ISSUE TITLE
[DOC] detection notebook - move plotting functions to `annotation.plotting`

### ISSUE THREAD

**Issue by fkiraly** [sktime/sktime#7285]
In the detection notebook 7, @alex-jg3 has created a number of plotting functions that would be nice to have generally.

I think we should refactor the plotting functions and move them to `annotation.plotting`. There are also some existing functions in the latter, it might be useful to see if they can be merged.

As a recipe:

1. look at every block in the notebook 7, where `matplotlib` is used and something plotted
2. try to turn this into a function `plot_something(param1, param2, ax=None)` (where if `ax` is passed, no new `ax` is created but that is used)
3. put the function in `annotation.plotting`, and have the notebook import it from there instead of defining the plot in the notebook.

**Comment by fkiraly** [comment]
Sure! Feel free to ask on discord dev-chat if you need any advice.

**Comment by PranavBhatP** [comment]
Hi @fkiraly, I'm new to sktime. I think this would be good issue for me to start with...Can I go ahead?

**Comment by PranavBhatP** [comment]
Hi @fkiraly, I'm new to sktime and this issue would be a good starting point for me. Can I work on a PR?

**Comment by fkiraly** [comment]
@PranavBhatP, I believe @sank8-2 has already taken on this issue and made a pull request - how about another?

**Comment by kartikbhtt7** [comment]
Hi @fkiraly @sank8-2, is this still being worked on? Would it be okay if I take a shot at implementing the required changes for #7300?

**Comment by fkiraly** [comment]
The PR https://github.com/sktime/sktime/pull/7300 looks stalled, so I think it can be worked on again. You could start from scratch, or based on https://github.com/sktime/sktime/pull/7300, i nboth cases @sank8-2 would be credited, too, for their partial work.

**Comment by hridaya14** [comment]
Hi @fkiraly , I’d love to contribute to this issue if it’s still open. Please let me know if I can take it up. Thanks!

**Comment by fkiraly** [comment]
yes, I think it is open again. The PR https://github.com/sktime/sktime/pull/7300 is a good start but tests are failling.

**Comment by hridaya14** [comment]
Hi @fkiraly ,

As per the requirements, I’ve moved the functions as specified in [this commit] https://github.com/sktime/sktime/commit/573ea9616fd274ea84aa581cd69f9fea20089fc2). I’d like to confirm if I’m on the right track and if the changes look fine before proceeding with writing tests and creating a PR.

Please let me know if any adjustments are needed. Looking forward to your feedback!

Thanks!

**Comment by shyamsharmas124-commits** [comment]
Hi @fkiraly, I see that PR #7300 is currently stalled with failing tests.

Would it be okay if I continue from that work or start a fresh implementation to move the plotting functions into `detection.plotting`? I’d be happy to help move this forward.
